lib/pointer.js

```diff
@@ -171,7 +174,7 @@ Pointer.parse = function (path) {
   }
 
   if (pointer[0] !== "") {
-    throw ono.syntax(`Invalid $ref pointer "${pointer}". Pointers must begin with "#/"`);
+    throw new GenericError(`Invalid $ref pointer "${pointer}". Pointers must begin with "#/"`);
```

Comment:
I believe we could have a dedicated `InvalidPointerError` here. Feels a tad weird to have a `GenericError` used here.
What do you think?